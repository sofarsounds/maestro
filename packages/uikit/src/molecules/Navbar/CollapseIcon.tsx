import React from 'react';
import styled, { css, keyframes } from '../../lib/styledComponents';

interface Props {
  onClick: () => void;
  open?: boolean;
}

const time = '0.15s';
const delay = '.25s';
const cubic = 'cubic-bezier(.25, .73, .85, .56)';
const height = 30;

const animate = keyframes`
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(45deg);
  }
`;

const animateUp = keyframes`
  0% {
    transform: rotate(0deg);
  }
100% {
    transform: rotate(-45deg);
}
`;

const Line = styled.span`
  display: block;
  position: absolute;
  height: 2px;
  width: 20px;
  background: #fff;
  border-radius: 1px;
  transition: all ${time} ${cubic};
  opacity: 1;
  left: 5px;
`;

const Hamburger = styled.div<Props>`
  width: ${height}px;
  height: ${height}px;
  position: relative;
  cursor: pointer;
  box-sizing: content-box;

  ${({ theme }) => theme.media.md`
    display: none;
  `}

  ${Line}:nth-child(1) {
    top: ${height / 2 - 3}px;
  }

  ${Line}:nth-child(2) {
    top: ${height / 2 + 2}px;
  }

  ${({ open }) =>
    open &&
    css`
      ${Line}:nth-child(1) {
        top: ${height / 2}px;
        transform-origin: center;
        animation: ${animate} ${time} ${cubic} ${delay} 1 normal forwards;
      }

      ${Line}:nth-child(2) {
        top: ${height / 2}px;
        transform-origin: center;
        animation: ${animateUp} ${time} ${cubic} ${delay} 1 normal forwards;
      }
    `}
`;

const CollapsibleIcon: React.SFC<Props> = ({ onClick, open }) => (
  <Hamburger onClick={onClick} open={open}>
    <Line />
    <Line />
  </Hamburger>
);

export default CollapsibleIcon;
