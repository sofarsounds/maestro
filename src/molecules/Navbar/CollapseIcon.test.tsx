import React from 'react';
import { mountWithTheme } from '../../test';

import CollapseIcon from './CollapseIcon';

const mockClick = jest.fn();

describe('Navbar <CollapseIcon />', () => {
  it('renders correctly in closed state', () => {
    const wrapper = mountWithTheme(<CollapseIcon onClick={mockClick} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when in open state', () => {
    const wrapper = mountWithTheme(<CollapseIcon onClick={mockClick} open />);

    expect(wrapper).toMatchSnapshot();
  });

  it('is clickable', () => {
    const wrapper = mountWithTheme(<CollapseIcon onClick={mockClick} open />);
    wrapper.simulate('click');

    expect(mockClick).toHaveBeenCalled();
  });
});
