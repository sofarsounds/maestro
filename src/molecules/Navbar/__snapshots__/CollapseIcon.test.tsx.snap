// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar <CollapseIcon /> renders correctly in closed state 1`] = `
.c2 {
  display: block;
  position: absolute;
  height: 2px;
  width: 20px;
  background: #fff;
  border-radius: 1px;
  -webkit-transition: all 0.15s cubic-bezier(.25,.73,.85,.56);
  transition: all 0.15s cubic-bezier(.25,.73,.85,.56);
  opacity: 1;
  left: 5px;
}

.c0 {
  width: 30px;
  height: 30px;
  position: relative;
  cursor: pointer;
  box-sizing: content-box;
}

.c0 .c1:nth-child(1) {
  top: 12px;
}

.c0 .c1:nth-child(2) {
  top: 17px;
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

<Component
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "default": "2px",
        "input": "2px",
        "large": "8px",
      },
      "colors": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueChristmas": "#3498DB",
        "blueMagic": "#34495E",
        "blueSmoke": "#949494",
        "brownSugar": "#9B525F",
        "checkBerry": "#98035A",
        "darkGrey": "#535353",
        "darkSmoke": "#767676",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "greenGarden": "#19BC9B",
        "greenOnions": "#8BC34B",
        "lightBlueSmoke": "#94949499",
        "macyGrey": "#DBDBDB",
        "mellowYellow": "#F1C410",
        "orangeCrush": "#F39C12",
        "paintItBlack": "#333333",
        "peaches": "#F99973",
        "pinkTriangle": "#E91E63",
        "primary": "#039842",
        "purpleHaze": "#3F51B5",
        "purplePills": "#9B59B6",
        "redRedWine": "#EF4423",
        "royalBlood": "#FF5000",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
        "yellowSubmarine": "#FFD426",
      },
      "dimensions": Object {
        "button": Object {
          "default": Object {
            "height": "44px",
            "padding": "0 25px",
          },
          "small": Object {
            "height": "40px",
            "padding": "0 15px",
          },
        },
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
        "navbarHeight": Object {
          "lg": "84px",
          "md": "74px",
          "xs": "62px",
        },
        "radioBox": "20px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "formGroupLabel": "14px",
        "h1": Object {
          "lg": "48px",
          "md": "34px",
          "xs": "24px",
        },
        "h2": Object {
          "lg": "34px",
          "md": "24px",
          "xs": "20px",
        },
        "h3": Object {
          "lg": "24px",
          "md": "20px",
          "xs": "16px",
        },
        "h4": Object {
          "lg": "20px",
          "md": "16px",
          "xs": "16px",
        },
        "h5": Object {
          "lg": "16px",
          "md": "14px",
          "xs": "14px",
        },
        "h6": Object {
          "lg": "14px",
          "md": "14px",
          "xs": "14px",
        },
        "overline": "10px",
        "title": Object {
          "lg": "54px",
          "md": "38px",
          "xs": "26px",
        },
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
      "zIndex": Object {
        "navbar": 100,
      },
    }
  }
>
  <CollapsibleIcon
    onClick={[MockFunction]}
  >
    <styled.div
      onClick={[MockFunction]}
    >
      <div
        className="c0"
        onClick={[MockFunction]}
      >
        <styled.span>
          <span
            className="c1 c2"
          />
        </styled.span>
        <styled.span>
          <span
            className="c1 c2"
          />
        </styled.span>
      </div>
    </styled.div>
  </CollapsibleIcon>
</Component>
`;

exports[`Navbar <CollapseIcon /> renders correctly when in open state 1`] = `
.c2 {
  display: block;
  position: absolute;
  height: 2px;
  width: 20px;
  background: #fff;
  border-radius: 1px;
  -webkit-transition: all 0.15s cubic-bezier(.25,.73,.85,.56);
  transition: all 0.15s cubic-bezier(.25,.73,.85,.56);
  opacity: 1;
  left: 5px;
}

.c0 {
  width: 30px;
  height: 30px;
  position: relative;
  cursor: pointer;
  box-sizing: content-box;
}

.c0 .c1:nth-child(1) {
  top: 12px;
}

.c0 .c1:nth-child(2) {
  top: 17px;
}

.c0 .c1:nth-child(1) {
  top: 15px;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hfqXXB 0.15s cubic-bezier(.25,.73,.85,.56) .25s 1 normal forwards;
  animation: hfqXXB 0.15s cubic-bezier(.25,.73,.85,.56) .25s 1 normal forwards;
}

.c0 .c1:nth-child(2) {
  top: 15px;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: fyuDSs 0.15s cubic-bezier(.25,.73,.85,.56) .25s 1 normal forwards;
  animation: fyuDSs 0.15s cubic-bezier(.25,.73,.85,.56) .25s 1 normal forwards;
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

<Component
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "default": "2px",
        "input": "2px",
        "large": "8px",
      },
      "colors": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueChristmas": "#3498DB",
        "blueMagic": "#34495E",
        "blueSmoke": "#949494",
        "brownSugar": "#9B525F",
        "checkBerry": "#98035A",
        "darkGrey": "#535353",
        "darkSmoke": "#767676",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green800": "#008736",
        "green900": "#006721",
        "greenGarden": "#19BC9B",
        "greenOnions": "#8BC34B",
        "lightBlueSmoke": "#94949499",
        "macyGrey": "#DBDBDB",
        "mellowYellow": "#F1C410",
        "orangeCrush": "#F39C12",
        "paintItBlack": "#333333",
        "peaches": "#F99973",
        "pinkTriangle": "#E91E63",
        "primary": "#039842",
        "purpleHaze": "#3F51B5",
        "purplePills": "#9B59B6",
        "redRedWine": "#EF4423",
        "royalBlood": "#FF5000",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
        "yellowSubmarine": "#FFD426",
      },
      "dimensions": Object {
        "button": Object {
          "default": Object {
            "height": "44px",
            "padding": "0 25px",
          },
          "small": Object {
            "height": "40px",
            "padding": "0 15px",
          },
        },
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
        "navbarHeight": Object {
          "lg": "84px",
          "md": "74px",
          "xs": "62px",
        },
        "radioBox": "20px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "formGroupLabel": "14px",
        "h1": Object {
          "lg": "48px",
          "md": "34px",
          "xs": "24px",
        },
        "h2": Object {
          "lg": "34px",
          "md": "24px",
          "xs": "20px",
        },
        "h3": Object {
          "lg": "24px",
          "md": "20px",
          "xs": "16px",
        },
        "h4": Object {
          "lg": "20px",
          "md": "16px",
          "xs": "16px",
        },
        "h5": Object {
          "lg": "16px",
          "md": "14px",
          "xs": "14px",
        },
        "h6": Object {
          "lg": "14px",
          "md": "14px",
          "xs": "14px",
        },
        "overline": "10px",
        "title": Object {
          "lg": "54px",
          "md": "38px",
          "xs": "26px",
        },
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
      "zIndex": Object {
        "navbar": 100,
      },
    }
  }
>
  <CollapsibleIcon
    onClick={[MockFunction]}
    open={true}
  >
    <styled.div
      onClick={[MockFunction]}
      open={true}
    >
      <div
        className="c0"
        onClick={[MockFunction]}
        open={true}
      >
        <styled.span>
          <span
            className="c1 c2"
          />
        </styled.span>
        <styled.span>
          <span
            className="c1 c2"
          />
        </styled.span>
      </div>
    </styled.div>
  </CollapsibleIcon>
</Component>
`;
